From 78c13394c917fd73ed8e3f0ad15462305b49c8b8 Mon Sep 17 00:00:00 2001
From: Marsell Kukuljevic <marsell@joyent.com>
Date: Tue, 6 Feb 2018 22:12:53 +1300
Subject: [PATCH] TRITON-108 SAPI dies when files missing in
 /sapi/sapi_manifests Reviewed by: Trent Mick <trent.mick@joyent.com> Approved
 by: Trent Mick <trent.mick@joyent.com>

---
 lib/server/stor/moray_local.js | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/lib/server/stor/moray_local.js b/lib/server/stor/moray_local.js
index 6d65a47..b97e31b 100644
--- a/lib/server/stor/moray_local.js
+++ b/lib/server/stor/moray_local.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -195,10 +195,14 @@ function loadAppObjects(uuid, stor, cb) {
             vasync.forEachParallel({
                 'inputs': muuids,
                 'func': fManifests
-            }, function (err, res) {
-                _.manifests = res.successes.map(function (m) {
+            }, function vasyncCb(err, res) {
+                _.manifests = res.successes.filter(function filterCb(m) {
+                    // if a file isn't found
+                    return (m !== null);
+                }).map(function mapCb(m) {
                     return (m.value);
                 });
+
                 return (subcb(err));
             });
         }
-- 
2.21.0

